qa/integration-tests/src/test/java/io/camunda/zeebe/broker/it/network/SmallMessageSizeTest.java

```diff
@@ -27,78 +28,72 @@
   private static final DataSize MAX_MESSAGE_SIZE = DataSize.ofKilobytes(12);
   private static final String LARGE_TEXT =
       "x".repeat((int) (MAX_MESSAGE_SIZE.toBytes() / VARIABLE_COUNT));
+  private static final String JOB_TYPE = "acme";
 
   private static final EmbeddedBrokerRule BROKER_RULE =
-      new EmbeddedBrokerRule(b -> b.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE));
+      new EmbeddedBrokerRule(
+          b -> {
+            b.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE);
+            b.getGateway().getLongPolling().setEnabled(false);
+          });
   private static final GrpcClientRule CLIENT_RULE = new GrpcClientRule(BROKER_RULE);
-  private static final String JOB_TYPE = "acme";
 
   @ClassRule
   public static RuleChain ruleChain = RuleChain.outerRule(BROKER_RULE).around(CLIENT_RULE);
 
-  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
-
-  private static BpmnModelInstance process(final String jobType) {
-    return Bpmn.createExecutableProcess("process")
-        .startEvent()
-        .serviceTask("task", t -> t.zeebeJobType(jobType))
-        .endEvent()
-        .done();
-  }
-
-  @Test
-  public void shouldSkipJobsThatExceedMessageSize() {
-    // given (two processes, the first has variables too big to fit into a message, the second fits
-    // into a message)
-    final var processDefinitionKey = CLIENT_RULE.deployProcess(process(JOB_TYPE));
-
-    // process with variables that are greater than the message size
-    final var processInstanceKey1 = CLIENT_RULE.createProcessInstance(processDefinitionKey);
-
+  private long processInstanceKey;
+
+  @Before
+  public void givenProcessWithLargeVariables() {
+    // given a process with a job
+    final var processDefinitionKey =
+        CLIENT_RULE.deployProcess(
+            Bpmn.createExecutableProcess("process")
+                .startEvent()
+                .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+                .endEvent()
+                .done());
+
+    processInstanceKey = CLIENT_RULE.createProcessInstance(processDefinitionKey);
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .exists();
+
+    // with variables that are greater than the message size
     for (int i = 0; i < VARIABLE_COUNT; i++) {
       CLIENT_RULE
           .getClient()
-          .newSetVariablesCommand(processInstanceKey1)
-          .variables(Map.of(String.valueOf(i), LARGE_TEXT))
+          .newSetVariablesCommand(processInstanceKey)
+          .variables(Map.of(String.format("large-%d", i), LARGE_TEXT))
           .send()
           .join();
     }
+    assertThat(
```

Comment:
Do we need to use an assertion here? I am always weary about assertions in setup code. We could also use awaitility and wait for the condition to become true. I think if it fails, I think we would get a timeout exception and not an assertion error. Would be worth trying out.

Reply:
Tried it out with an aliased await and it works :)

```java
Awaitility.await("large variables CREATED")
        .until(
            () ->
                RecordingExporter.variableRecords(VariableIntent.CREATED)
                    .withProcessInstanceKey(processInstanceKey)
                    .filter(v -> v.getValue().getName().startsWith("large"))
                    .limit(4)
                    .count(),
            n -> n == 5);
```

```
org.awaitility.core.ConditionTimeoutException: Condition with alias 'large variables CREATED' didn't complete within 10 seconds because lambda expression in io.camunda.zeebe.broker.it.network.SmallMessageSizeTest: expected the predicate to return <true> but it returned <false> for input of <4L>.

	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AbstractHamcrestCondition.await(AbstractHamcrestCondition.java:86)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:601)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:618)
	at io.camunda.zeebe.broker.it.network.SmallMessageSizeTest.givenProcessWithLargeVariables(SmallMessageSizeTest.java:76)
```